### PR TITLE
Support transforming enum values to internal representation

### LINKED ISSUE #39: Support custom internal representation of enums
Currently Ariadne supports enums that are represented as strings that are mapped 1:1 to what they are in SDL. This should take care of majority of use cases, but there may be edge cases and legacy systems, where enum is defined for existing value that may be integer, and updating python implementation is out of question.

[Apollo doc for reference.](https://www.apollographql.com/docs/apollo-server/features/scalars-enums.html#internal-values)

### PR DESCRIPTION
**Pull request by rafalp**
This PR adds `Enum` *bindable* that takes name of enum in schema and dict of mappings, enabling easy transformation of enum values between GraphQL and internal representation:

```
role = Enum(
    "role",
    {
        "ADMIN": 1,
        "MODERATOR": 2,
        "MVP": 4,
        "REGULAR" 5,
        "PROBATION": 6,
    }
)
```

Fixes #39

### CODE REVIEW

**Comment by rafalp** [comment]
Note to self: passing actuall `Enum` as `ariadne.Enum` value should be possible too:

```
my_enum = MyEnum(Enum):
    RED = 1
    BLUE = 2


graphql_enum = ariadne.Enum("SomeEnum", my_enum)
```

**Comment by rafalp** [comment]
And now you can also pass either `enum.Enum` or `enum.IntEnum` to `ariadne.Enum`.

File: tests/test_enums.py [hunk 1]
```diff
@@ -45,7 +49,7 @@ def test_unsuccesfull_invalid_enum_value_evaluation():
 """
 
 
-def test_succesfull_enum_value_passed_as_argument():
+def test_successfull_enum_value_passed_as_argument():
```
Comment (patrys) [inline tests/test_enums.py]:
```suggestion
def test_successful_enum_value_passed_as_argument():
```

File: tests/test_enums.py [hunk 2]
```diff
@@ -54,10 +58,112 @@ def test_succesfull_enum_value_passed_as_argument():
     assert result.errors is None, result.errors
 
 
-def test_unsuccesfull_invalid_enum_value_passed_as_argument():
+def test_unsuccessfull_invalid_enum_value_passed_as_argument():
```
Comment (patrys) [inline tests/test_enums.py]:
```suggestion
def test_unsuccessful_invalid_enum_value_passed_as_argument():
```

**Pull request merged**